File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        newNode = TreeNode(val)
        if not root: return newNode
        cur = root
        while cur:
            if cur.val > val:
                if cur.left:
                    cur = cur.left
                else:
                    cur.left = newNode
                    break
            else:
                if cur.right:
                    cur = cur.right
                else:
                    cur.right = newNode
                    break
        return root

File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        TreeNode* node = new TreeNode(val);
        if (root == NULL) return node;
        TreeNode* cur = root;
        while (cur != NULL) {
            if (cur->val > val) {
                if (cur->left != NULL) cur = cur->left;
                else {
                    cur->left = node;
                    break;
                }
            };
            if (cur->val < val) {
                if (cur->right != NULL) cur = cur->right;
                else {
                    cur->right = node;
                    break;
                }
            };
        }
        return root;
    }
};